edge-util/src/Microsoft.Azure.Devices.Edge.Util/PersistentStorageValidation.cs

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Util
+{
+    using System.IO;
+    using Microsoft.Extensions.Logging;
+    using Newtonsoft.Json;
+
+    public static class PersistentStorageValidation
+    {
+        public static void ValidateStorageIdentity(string storagePath, string deviceId, string iotHubHostname, string moduleId, Option<string> moduleGenerationId)
+        {
+            ILogger logger = SetupLogger();
+            DeviceIdentity currentIdentity = new DeviceIdentity(deviceId, iotHubHostname, moduleId, moduleGenerationId);
+            string filepath = Path.Combine(storagePath, "DEVICE_IDENTITY");
+            string json = JsonConvert.SerializeObject(currentIdentity);
+
+            if (!File.Exists(filepath))
+            {
+                File.WriteAllText(filepath, json);
+            }
+            else
+            {
+                DeviceIdentity savedIdentity = JsonConvert.DeserializeObject<DeviceIdentity>(File.ReadAllText(filepath));
+
+                if (currentIdentity.Equals(savedIdentity))

```

Comment:
should this be not equal, based on the log message?

Reply:
Yes, good catch. Thank you!